src/train.py

```diff
@@ -78,9 +80,63 @@ def main(operation, data_dir, output_dir, model, model_fn, in_weights_path=None,
                             num_parallel_calls=tf.data.AUTOTUNE)
                      .repeat())
 
-    # load weights if the model is supposed to do so
+    # load weights if the model is supposed to do so (i.e. fine-tune mode)
     if operation == 'fine-tune':
-        model.load_weights(in_weights_path)
+        # if input or output dimension changed w.r.t pretrained model
+        if finetune_old_inp_dim or finetune_old_out_dim:
+            if model == "U-Net":
+                # set dimensions for creating pretrained model
```

Comment:
Maybe you could put this block in a function/method, so that the code get more structured?